sklearn/cluster/_birch.py

```diff
@@ -5,9 +5,11 @@
 
 import warnings
 import numpy as np
+import sklearn
 from scipy import sparse
 from math import sqrt
 
+import sklearn.metrics.pairwise
```

Comment:
```suggestion
from ..metrics import pairwise_distances_argmin
```

We use relative imports for internal references to sklearn

Reply:
Right. The way I did the import here is  not inline with the practice. I can surely revert. 
I had a reason for this. It is related to the performance script. I wanted to profile pairwise.euclidean_distances to see memory during chunk processing
But I did not want to decorate the function with @profile each time I run the performance test. 
And I surely cannot add this decoration permanently in the source code.
So I found a way to decorate at runtime.  From the performance script:

```
import sklearn.cluster._birch as br
....
birch.predict = profile(birch.predict)
        br.sklearn.metrics.pairwise.PAIRWISE_DISTANCE_FUNCTIONS['euclidean'] \
            = profile(br.sklearn.metrics.pairwise.PAIRWISE_DISTANCE_FUNCTIONS['euclidean'])

```
But in order to make sure that I decorate the same instance function of euclidean_distances() that is callled during 
`pairwise.distances_argmin()`
I need this import in _birch.py:  
`import sklearn.metrics.pairwise
`
In short it was  so we will be able to run the performance test with touching the source code ( with @profile decoration ). But, if required  I can remove it

Reply:
It's perfectly fine during testing and profiling but for inclusion in the codebase it will have to follow our coding practices. Btw, I think you can switch the tag to MRG. Your code is ready for review and it will attract more reviewers :)

Reply:
OK. will do